README.md

```diff
@@ -243,91 +256,381 @@ The SDK provides a simple interface using the `.upload()` method to upload files
 
 Accepts an object specifying the parameters to be used to list and search files. All parameters specified in the [documentation here](https://docs.imagekit.io/api-reference/media-api/list-and-search-files) can be passed as it is with the correct values to get the results.
 
 ```cs
-List<ListAPIResponse> resp = await imagekit
-    .Skip(0)
-    .Limit(10)
-    .ListFilesAsync();
+GetFileListRequest ob = new GetFileListRequest();
+ob.limit = 10;
+ob.skip = 0;
+ResultList resp = await imagekit.GetFileListRequestAsync(ob);
+```
+
+**2\. Get File Details**
+
+Accepts the file ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-details).
+
+```cs
+Result resp = await imagekit.GetFileDetailsAsync(fileId);
 ```
 
-You can also use advanced [searching](https://docs.imagekit.io/api-reference/media-api/list-and-search-files#advanced-search-queries) and sorting using `SearchQuery` and `Sort` function.
+**3\. Get File Versions**
+
+Accepts the file ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-versions).
 
 ```cs
-List<ListAPIResponse> resp = await imagekit
-    .Skip(0)
-    .Limit(3)
-    .Sort("DESC_SIZE")
-    .SearchQuery("tags IN [\"tag1\"]")
-    .ListFilesAsync();
+String fileId = "file-id-1";
+ResultFileVersions resultFileVersions = imageKit.getFileVersions(fileId);
 ```
 
-**2. Get File Details**
+**4\. Get File Version details**
 
-Accepts the file ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-details).
+Accepts the file ID and version ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-version-details).
+
+```cs
+String fileId = "file-id-1";
+String versionId = "file-version-id-1";
+ResultFileVersionDetails resultFileVersionDetails = imageKit.getFileVersionDetails(fileId, versionId);
+```
+
+**5\. Update File Details**
+
+Accepts an object of class `FileUpdateRequest` specifying the parameters to be used to update file details. All parameters specified in the \[documentation here\] (https://docs.imagekit.io/api-reference/media-api/update-file-details) can be passed via their setter functions to get the results.
+
+```cs
+List<String> tags = new ArrayList<>();
+tags.add("tag-1");
+tags.add("tag-2");
+tags.add("tag-3");
+
+List<String> aiTags = new ArrayList<>();
+aiTags.add("ai-tag-1");
+FileUpdateRequest fileUpdateRequest = new FileUpdateRequest("fileId");
+fileUpdateRequest.Tags = tags;
+fileUpdateRequest.AITags = aiTags;
+
+UpdateFileDetail result = imageKit.updateFileDetail(fileUpdateRequest); 
+```
+
+**6\. Delete File**
+
+Accept the file ID and delete a file as per the [API documentation here](https://docs.imageKit.io/api-reference/media-api/delete-file).
+
+```cs
+String fileId = "file-id-1";
+Result result = imageKit.deleteFile(fileId);
```

Comment:
`Result` type is used for file detail as well. But delete file API response is empty. Is this correct?